src/artery/envmod/GlobalEnvironmentModel.cc

```diff
@@ -340,7 +354,9 @@ GlobalEnvironmentModel::preselectObjects(const std::string& ego, const std::vect
     ObjectRtree::const_query_iterator it = query_intersections(mObjectRtree, area);
     for (; it != mObjectRtree.qend(); ++it) {
         if (it->second->getExternalId() != ego) {
-            objectsInSearchArea.push_back(it->second);
+            if (it->second->isVisible()) {
```

Comment:
I would merge this check with `getExternalId() != ego`. But that's not a big deal.

Reply:
Done :+1: